addons/project/upgrades/1.4/pre-migrate.py

```diff
@@ -0,0 +1,19 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+def migrate(cr, version):
+    cr.execute("""
+        UPDATE ir_rule r
+           SET domain_force = '[
+                ("project_id.privacy_visibility", "=", "portal"),
+                ("active", "=", True),
+                "|",
+                    ("message_partner_ids", "child_of", [user.partner_id.commercial_partner_id.id]),
+                    ("project_id.collaborator_ids.partner_id", "=", user.partner_id.id),
+            ]'
+          FROM ir_model_data d
+         WHERE d.res_id = r.id
+           AND d.model = 'ir.rule'
+           AND d.module = 'project'
+           AND d.name = 'project_task_rule_portal_project_sharing'
```

Comment:
We should update to the new domain if the old domain matches, as mentioned in this -https://github.com/odoo/odoo/pull/228590#discussion_r2428331462.